arm/examples/sdk.go

```diff
@@ -0,0 +1,168 @@
+	}
+	tenantID := os.Getenv("AZURE_TENANT_ID")
+	if tenantID == "" {
+		return nil, fmt.Errorf("empty $AZURE_TENANT_ID")
+	}
+
+	// Here we build the SDK based on a user's configuration.
+	sdk := &SDK{
+		Authentication: &Authentication{
+			ClientID:       clientID,
+			ClientSecret:   clientSecret,
+			SubscriptionID: subscriptionID,
+			TenantID:       tenantID,
+			HashMap: map[string]string{
+				"AZURE_CLIENT_ID":       clientID,
+				"AZURE_CLIENT_SECRET":   clientSecret,
+				"AZURE_SUBSCRIPTION_ID": subscriptionID,
+				"AZURE_TENANT_ID":       tenantID,
+			},
+		},
+	}
+
+	// Generate an authenticated token.
+	authenticatedToken, err := helpers.NewServicePrincipalTokenFromCredentials(sdk.Authentication.HashMap, azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	sdk.Authentication.AuthenticatedToken = authenticatedToken
+	return sdk, nil
+}
+
+// E will check if an error is nil, and exit the program in the case of an error.
+// Use this as convenience to simplify code in various examples.
+func E(err error) {
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(2)
+	}
+}
+
+// S will take an arbitrary string, and return a newly created pointer to this string.
+// Use this as convenience to simplify code in various examples.
+func S(i string) *string {
+	return &i
+}
+
+// I32 will take an arbitrary int, and return a newly created pointer to an int32.
+// Use this as convenience to simplify code in various examples.
+func I32(i int) *int32 {
+	i32 := int32(i)
+	return &i32
+}
+
+// F64 will take an arbitrary int, and return a newly created pointer to a float64.
+// Use this as convenience to simplify code in various examples.
+func F64(i int) *float64 {
+	f64 := float64(i)
+	return &f64
+}
+
+// PrettyJSON will take an arbitrarry interface{} and attempt to print the data structure out to STDOUT.
+// Use this as convenience to simplify code in various examples.
+func PrettyJSON(data interface{}) error {
+	buffer := new(bytes.Buffer)
+	encoder := json.NewEncoder(buffer)
+	encoder.SetIndent("", "\t")
+	err := encoder.Encode(data)
+	if err != nil {
+		return err
+	}
+	fmt.Println(buffer.String())
+	return nil
+}
+
+// EnsureResourceGroup will ensure that a resource group of an arbitrary name and location exists.
+// Use this as convenience to simplify code in various examples.
+func (sdk *SDK) EnsureResourceGroup(resourceGroupName, location string) error {
+	resourceGroup := resources.NewGroupsClient(sdk.Authentication.SubscriptionID)
+	resourceGroup.Authorizer = autorest.NewBearerAuthorizer(sdk.Authentication.AuthenticatedToken)
+
+	// ---------------------------------
+	// Resource Group Parameters
+	parameters := resources.Group{
+		Name:     &resourceGroupName,
+		Location: &location,
+	}
+
+	newResourceGroup, err := resourceGroup.CreateOrUpdate(resourceGroupName, parameters)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Ensured resource group [%s] exists\n", *newResourceGroup.Name)
```

Comment:
Again, I'm not a big fan of directly printing to Stdout inside of a helper function.